docs: move description of host_language from Macros to Language-specific Keycodes (#24584)
qmk/qmk_firmware: docs/reference_keymap_extras.md

Keyboards are able to support a wide range of languages. However, this support is not actually achieved within the keyboard itself - instead, it sends numerical codes, which the operating system maps to the appropriate characters depending on the user's configured keyboard layout. By default (and per the HID spec), this is the US ANSI layout. For example, when a Swedish person presses the key with the `å` character printed on it, the keyboard is *actually* sending the keycode for `[`.

Obviously, this can get confusing, so QMK provides language-specific keycode aliases for many keyboard layouts. These are used in place of the `KC_` prefixed ones. They won't do much on their own - you still have to set the matching keyboard layout in your OS settings. Think of them more as keycap labels for your keymap. The language-specific keycode aliases are defined in the files listed in the [Keycodes Header](#header-files) column below.

## Selecting Your Host Keyboard Layout

> To select a host keyboard layout, simply `#include` one of the [keycode headers](#header-files) below at the top of your `keymap.c`.

Reviewer: ````suggestion
To select a host keyboard layout, simply `#include` one of the [keycode headers](#header-files) below at the top of your `keymap.c`. Example:
```c
#include keymap_japanese.h
```
````
Include C example.

Author: Added an additional newline surrounding the codeblock

Alternatively, if using `keymap.json`, add the `host_language` key as shown in the following example. The available languages are those with a _Sendstring LUT Header_ entry in one of the [Header Files](#header-files) tables.

```json
{
    "keyboard": "handwired/my_macropad",
    "keymap": "my_keymap",

>     "host_language": "dvorak",
>     "macros": [
>         ["Hello, World!"]
>     ],
>     "layout": "LAYOUT_all",
>     "layers": [
>         ["QK_MACRO_0"]

Reviewer: ```suggestion
    "host_language": "swedish",
    "layout": "LAYOUT_all",
    "layers": [
        ["SE_A"]
```
Avoid conflating a "macros" block here to minimise end-user confusion because that block will not be present in a default `keymap.json`. Use a language specific example and keycode the illustrate key code use-case.

Author: Thanks, very nice to have the `SE_` example!!
Should use `SE_ARNG` so that it fits the example in text (å), or is that confusing?

Reviewer: I think `SE_ARNG` is also a fine example.

Keyboards are able to support a wide range of languages. However, this support is not actually achieved within the keyboard itself - instead, it sends numerical codes, which the operating system maps to the appropriate characters depending on the user's configured keyboard layout. By default (and per the HID spec), this is the US ANSI layout. For example, when a Swedish person presses the key with the `å` character printed on it, the keyboard is *actually* sending the keycode for `[`.

Obviously, this can get confusing, so QMK provides language-specific keycode aliases for many keyboard layouts. These are used in place of the `KC_` prefixed ones. They won't do much on their own - you still have to set the matching keyboard layout in your OS settings. Think of them more as keycap labels for your keymap. The language-specific keycode aliases are defined in the files listed in the [Keycodes Header](#header-files) column below.

## Selecting Your Host Keyboard Layout

To select a host keyboard layout, simply `#include` one of the [keycode headers](#header-files) below at the top of your `keymap.c`. Example:

```c

> #include keymap_japanese.h

Reviewer 2: ```suggestion
#include "keymap_japanese.h"
```

> #include "keymap_japanese.h"

Reviewer 2: ```suggestion
#include QMK_KEYBOARD_H

#include "keymap_japanese.h" // [!code focus]
```
On further thought this may help too.

```

Alternatively, if using `keymap.json`, add the `host_language` key as shown in the following example. The available languages are those with a _Sendstring LUT Header_ entry in one of the [Header Files](#header-files) tables.

```json
{
    "keyboard": "handwired/my_macropad",
    "keymap": "my_keymap",

>     "host_language": "swedish",

Reviewer 2: ```suggestion
    "host_language": "swedish", // [!code focus]
```
This will blur out the surrounding lines to draw attention to this one in particular. Could also do diff-style with `// [!code ++]`.

Author: ![image](https://github.com/user-attachments/assets/3dda1a28-f095-4b7a-ac83-3ed25307bea1)

Obviously, this can get confusing, so QMK provides language-specific keycode aliases for many keyboard layouts. These won't do much on their own - you still have to set the matching keyboard layout in your OS settings. Think of them more as keycap labels for your keymap.

> Simply `#include` one of the keycode headers below at the top of your `keymap.c`, or set the `host_language` field in your `keymap.json` to the corresponding language. Then assign the keycodes defined in the header in place of the `KC_` prefixed ones.

Reviewer: ````suggestion
Simply `#include` one of the keycode headers below at the top of your `keymap.c`, and assign the keycodes defined in the header in place of the `KC_` prefixed ones.

If your keymap is setup using a `keymap.json` file, this can be accomplished using the `host_language` key in that file. Example:
```json
{
    "keyboard": "handwired/my_macropad",
    "keymap": "my_keymap",
    "host_language": "swedish",
    "layout": "LAYOUT_all",
    "layers": [
        ["SE_A"]
    ]
}
```
````
An example will help remove ambiguity with the syntax.

Author: As the example already exists on the Macros page, should we just move it into here and reference it in macros? See https://github.com/qmk/qmk_firmware/pull/24584/commits/bbe17cfebb8653bb40769d25155b6716a86542f4
